certbot-apache/certbot_apache/dualparser.py

```diff
@@ -0,0 +1,99 @@
+""" Tests for ParserNode interface """
+from certbot_apache import assertions
+from certbot_apache import augeasparser
+
+
+class DualNodeBase(object):
+    """ Dual parser interface for in development testing. This is used as the
+    base class for dual parser interface classes. This class handles runtime
+    attribute value assertions."""
+
+    def save(self, msg):  # pragma: no cover
```

Comment:
This would be good if there would be a bunch of these, but there just really is `save` and `set_parameters`. One of the following PR's will bring along also `delete_child` and `unsaved_files`, but these unfortunately require different test assertions, so it would not be feasible to generalize them all. The current way keeps the code a bit more readable.